src/alias/azext_alias/argument.py

```diff
@@ -0,0 +1,165 @@
+    if len(pos_args_name) != len(set(pos_args_name)):
+        raise CLIError(DUPLICATED_PLACEHOLDER_ERROR.format(arg))
+
+    for i, pos_arg_name in enumerate(pos_args_name):
+        if pos_arg_name.isdigit():
+            pos_args_name[i] = '_{}'.format(pos_arg_name)
+
+    return pos_args_name
+
+
+def stringify_placeholder_expr(arg):
+    """
+    Jinja does not accept numbers as placeholder names,
+    so add a "_" before the number to make them valid placeholder names.
+    Surround placeholders expressions with "" so we can preserve spaces inside the positional arguments.
+
+    Args:
+        arg: The string to process.
+
+    Returns:
+        A processed string where placeholders are surrounded by "" and
+        numbered placeholders are prepended with "_".
+    """
+    number_placeholders = list(map(str.strip, re.findall(NUMBER_PLACEHOLDER_REGEX, arg)))
+    for number_placeholder in number_placeholders:
+        number = re.search(r'\d+', number_placeholder).group()
+        arg = arg.replace(number_placeholder, '{{_' + number + '}}')
+
+    return arg.replace('{{', '"{{').replace('}}', '}}"')
+
+
+def build_pos_args_table(full_alias, args, start_index):
+    """
+    Build a dictionary of position argument.
+
+    Args:
+        full_alias: The full alias (including any placeholders).
+        args: The arguments that the user inputs in the terminal.
+        start_index: The index at which we start taking position arguments.
+    Returns:
+        A dictionary with the key beign the name of the placeholder and its value
+        being the respective positional argument.
+    """
+    pos_args_placeholder = get_pos_args_names(full_alias)
+    pos_args = args[start_index: start_index + len(pos_args_placeholder)]
+
+    if len(pos_args_placeholder) != len(pos_args):
+        error_msg = INSUFFICIENT_POS_ARG_ERROR.format(full_alias,
+                                                      len(pos_args_placeholder),
+                                                      '' if len(pos_args_placeholder) == 1 else 's',
+                                                      len(pos_args))
+        raise CLIError(error_msg)
+
+    # Escape '"' because we are using "" to surround placeholder expressions
+    for i, pos_arg in enumerate(pos_args):
+        pos_args[i] = pos_arg.replace('"', '\\"')
+
+    return dict(zip(pos_args_placeholder, pos_args))
+
+
+def render_template(cmd_derived_from_alias, pos_args_table):
+    """
+    Render cmd_derived_from_alias as a Jinja template with pos_args_table as the arguments.
+
+    Args:
+        cmd_derived_from_alias: The string to be injected with positional arguemnts.
+        pos_args_table: The dictionary used to rendered.
+    Returns:
+        A processed string with positional arguments injected.
+    """
+    try:
+        cmd_derived_from_alias = stringify_placeholder_expr(cmd_derived_from_alias)
+        template = Template(cmd_derived_from_alias)
```

Comment:
advice: to improve readability, I would have written this line as
```
template = jinja.Template(cmd_derived_from_alias)
```
This form makes it more obvious to the reader that a Jinja template is in play here.